pkg/ccl/backupccl/backup_processor.go

```diff
@@ -138,22 +158,37 @@ func runBackupProcessor(
 
 	// TODO(pbardea): Check to see if this benefits from any tuning (e.g. +1, or
 	//  *2). See #49798.
-	numSenders := int(kvserver.ExportRequestsLimit.Get(&settings.SV)) * 2
-	targetFileSize := storageccl.ExportRequestTargetFileSize.Get(&settings.SV)
+	numSenders := int(kvserver.ExportRequestsLimit.Get(&clusterSettings.SV)) * 2
+	targetFileSize := storageccl.ExportRequestTargetFileSize.Get(&clusterSettings.SV)
 
 	// For all backups, partitioned or not, the main BACKUP manifest is stored at
 	// details.URI.
 	defaultConf, err := cloudimpl.ExternalStorageConfFromURI(spec.DefaultURI, spec.User())
 	if err != nil {
 		return err
 	}
-	storageByLocalityKV := make(map[string]*roachpb.ExternalStorage)
+	defaultStore, err := flowCtx.Cfg.ExternalStorage(ctx, defaultConf)
```

Comment:
nit: could make this a var and only conditionally initialize it / defer closing if `writeSSTsInProcessor`